node/pkg/evm/connectors/poller.go

```diff
@@ -0,0 +1,218 @@
+package connectors
+
+import (
+	"context"
+	"fmt"
+	"math/big"
+	"time"
+
+	"github.com/certusone/wormhole/node/pkg/supervisor"
+	ethEvent "github.com/ethereum/go-ethereum/event"
+
+	ethereum "github.com/ethereum/go-ethereum"
+	ethCommon "github.com/ethereum/go-ethereum/common"
+	ethHexUtils "github.com/ethereum/go-ethereum/common/hexutil"
+	"go.uber.org/zap"
+)
+
+type PollFinalizer interface {
+	IsBlockFinalized(ctx context.Context, block *NewBlock) (bool, error)
+}
+
+// BlockPollConnector polls for new blocks instead of subscribing when using SubscribeForBlocks. It allows to specify a
+// finalizer which will be used to only return finalized blocks on subscriptions.
+type BlockPollConnector struct {
+	Connector
+	DelayInMs           int
```

Comment:
Seems like we multiply this by `time.Millisecond` in the only place it's used so can this just be a `time.Duration` instead?

Reply:
Great point!